core/base/inc/TDirectory.h

```diff
@@ -304,6 +304,7 @@ can be replaced with the simpler and exception safe:
 
    static Bool_t       Cd(const char *path);
    static void         DecodeNameCycle(const char *namecycle, char *name, Short_t &cycle, const size_t namesize = 0);
+   R__DEPRECATED(6,30, "Cannot be used safely, use `name + ';' + std::to_string(cycle)`")
```

Comment:
What about introducing:
```
static void EncodeNameCycle(std::string &buffer, const char *name, Short_t cycle);
```

Reply:
I see no usage inside ROOT, and I GitHub shows that the only occurrence of `TDirectory::EncodeNameCycle` is within ROOT (and its clones), defining the function. I.e.: not needed; people have 1/2 year to complain :-)